crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -0,0 +1,193 @@
+    .expect("The current implementation resolved all futures instantly")?;
+
+    db_tx
+        .storage_as_mut::<DaCompressedBlocks>()
+        .insert(&block.header().consensus().height, &compressed)?;
+
+    Ok(())
+}
+
+struct CompressTx<'a, Tx> {
+    db_tx: &'a mut Tx,
+    block_events: &'a [Event],
+}
+
+macro_rules! impl_temporal_registry {
+    ($($name:ident: $type:ty),*$(,)?) => {
+        $(
+            impl<'a, Tx> TemporalRegistry<$type> for CompressTx<'a, Tx>
+            where
+                Tx: OffChainDatabaseTransaction,
+            {
+                fn read_registry(
+                    &self,
+                    key: fuel_core_types::fuel_compression::RegistryKey,
+                ) -> anyhow::Result<$type> {
+                    let v = self
+                        .db_tx
+                        .storage_as_ref::<DaCompressionTemporalRegistry>()
+                        .get(&(RegistryKeyspace::$name, key))?
+                        .ok_or(not_found!(DaCompressionTemporalRegistry))?
+                        .into_owned();
+                    match v {
+                        RegistryKeyspaceValue::$name(v) => Ok(v),
+                        _ => anyhow::bail!("Unexpected value in the registry"),
+                    }
+                }
+
+                fn write_registry(
+                    &mut self,
+                    key: fuel_core_types::fuel_compression::RegistryKey,
+                    value: $type,
+                ) -> anyhow::Result<()> {
+                    // Write the actual value
+                    self.db_tx
+                        .storage_as_mut::<DaCompressionTemporalRegistry>()
+                        .insert(&(RegistryKeyspace::$name, key), &RegistryKeyspaceValue::$name(value.clone()))?;
+
+                    // Remove the overwritten value from index, if any
+                    self.db_tx
+                        .storage_as_mut::<DaCompressionTemporalRegistryIndex>()
+                        .remove(&RegistryKeyspaceValue::$name(value.clone()))?;
+
+                    // Add the new value to the index
+                    self.db_tx
+                        .storage_as_mut::<DaCompressionTemporalRegistryIndex>()
+                        .insert(&RegistryKeyspaceValue::$name(value), &key)?;
+
+                    Ok(())
+                }
+
+                fn registry_index_lookup(
+                    &self,
+                    value: &$type,
+                ) -> anyhow::Result<Option<fuel_core_types::fuel_compression::RegistryKey>>
+                {
+                    Ok(self
+                        .db_tx
+                        .storage_as_ref::<DaCompressionTemporalRegistryIndex>()
+                        .get(&RegistryKeyspaceValue::$name(value.clone()))?
+                        .map(|v| v.into_owned()))
+                }
+            }
+
+        )*
+    };
+}
+
+// Arguments here should match the tables! macro from crates/compression/src/tables.rs
```

Comment:
It would be better if the implementation of the storage was independent of the `tables!` macro. It seems the removing of `RegistryKeyspace` and `RegistryKeyspaceValue` from implementation will solve that.

Reply:
It is independent. The comment here just tells the future person modifying the code where to look for values that are used here. But now it's removed in 946a2e8f23fef2109fb9ba286ebd2d451c2af7cb anyway.